docs/plugins.md

```diff
@@ -0,0 +1,207 @@
+# Plugins Guide
+Plugins are way to enhance the basic DataHub functionality in a custom manner.
+
```

Comment:
Seems that this doc needs a bit of work. 

TODO: We need to add a migration guide in here

Reply:
Yes right

Reply:
Updated the doc